src/generator/AutoRest.Ruby/Model/MethodRb.cs

```diff
@@ -606,6 +606,16 @@ public string GetDeserializationString(IModelType type, string valueReference =
                 builder.AppendLine("{1} = @client.deserialize(result_mapper, {0}, '{1}')", responseVariable, valueReference);
             }
 
+            if (type is CompositeType)
```

Comment:
Agreed that the resource group is an azure concept. Thanks for pointing that. I have changed the code with the latest commit